app/Http/Controllers/Consumables/ConsumableReplenishController.php

```diff
@@ -0,0 +1,165 @@
+            return redirect()->route('consumables.index')->with('error', trans('admin/consumables/message.not_found'));
+        }
+
+        $request->validate([
+            "checkout_qty" => "required|regex:/^[0-9]*$/|gt:0"
+          ],[            
+            'checkout_qty.gt' =>  trans('admin/consumables/message.under'),
+            'checkout_qty.required' => trans('admin/consumables/message.required'),
+            'checkout_qty.regex' => trans('admin/consumables/message.numeric'),
+          ]);
+
+        $this->authorize('update', $consumable);
+
+        $admin_user = Auth::user();
+
+        if ($request->hasFile('file')) {
+            if (! Storage::exists('private_uploads/consumables/docs')) {
+                Storage::makeDirectory('private_uploads/consumables/docs', 775);
+            }
+
+            $file = $request->file('file');
+            $extension =  $request->file('file')->getClientOriginalExtension();
+            $file_name = 'consumable_replenish-'.$consumable->id.'-'.str_random(8).'-'.str_slug(basename($file->getClientOriginalName(), '.'.$extension)).'.'.$extension;
+        
+            if ($extension=='svg') {
+                Log::debug('This is an SVG');
+
+                    $sanitizer = new Sanitizer();
+                    $dirtySVG = file_get_contents($file->getRealPath());
+                    $cleanSVG = $sanitizer->sanitize($dirtySVG);
+
+                    try {
+                        Storage::put('private_uploads/consumables/docs/'.$file_name, $cleanSVG);
+                    } catch (\Exception $e) {
+                        Log::debug('Upload no workie :( ');
+                        Log::debug($e);
+                    }
+            } else {
+            Storage::put('private_uploads/consumables/docs/'.$file_name, file_get_contents($file));
+            }
+            $consumable->logUpload($file_name, e($request->get('notes')));
+        } 
+
+        
+        // Update the consumable data
+        $consumable->assigned_to = e($request->input('assigned_to'));
+
+        isset($file_name) ? $file_name : $file_name = ''; 
+
+        $consumable->replenishusers()->attach($consumable->id, [
+            'consumable_id' => $consumable->id,            
+            'user_id' => $admin_user->id,
+            'initial_qty' => $consumable->qty,            
+            'total_replenish' => e($request->input('checkout_qty')),
+            'replenish_note' => e($request->input('replenish_note')),
+            'order_number'  => e($request->input('order_number')),
+            'file' => $file_name
+        ]);
+
+        $addedquantity = $consumable->qty + e($request->input('checkout_qty'));
+        
+        // Storing value to database
+        event(new ConsumableCheckedOut($consumable, Auth::user(), $consumable->qty, $request->input('note'), $request->input('checkout_qty')));
+
+        // Updating quantity to consumable
+        DB::table('consumables')
```

Comment:
Since we already have a `$consumable` Model element, we should be able to say `$consumable->qty = $addedquantity` and then `$consumable->save()` - though you could shorten that to `$consumable->update(['qty' => $addedquantity])` if you wanted. We just prefer to avoid using `DB::table()` constructs if there's any way we can.

Reply:
I will update it and check it further

Reply:
DB::table() constructs has been removed and updated